tests/integration/test_execution.py

```diff
@@ -592,6 +600,43 @@ def test_should_consume_message_without_correlation_id_when_it_is_not_required(s
     assert received_body == some_body
 
 
+def test_should_use_heartbeat_and_dont_lose_connection_when_using_background_processing(caplog, settings, mocker):
+    """
+    The listener in this code has a message handler that takes a long time to process a message (see
+    _test_callback_function_with_sleep_three_seconds function). Using a background process to process
+    the message, the heartbeat functionality should still work, maintaining the connection
+    estabilished between broker and consumer.
+    """
+    caplog.set_level(logging.DEBUG)
+    some_destination = f"some-lorem-destination-{uuid.uuid4()}"
+
+    # In order to publish sample data
+    with build_publisher().auto_open_close_connection() as publisher:
+        some_body = {"keyOne": 1, "keyTwo": 2}
+        publisher.send(some_body, some_destination, attempt=1)
+
+    settings.STOMP_OUTGOING_HEARTBIT = 1000
+    settings.STOMP_INCOMING_HEARTBIT = 1000
+    mocker.patch("django_stomp.execution.should_process_msg_on_background", True)
+
+    message_consumer = start_processing(
+        some_destination, myself_with_test_callback_sleep_three_seconds, is_testing=True, return_listener=True
+    )
+
+    wait_for_message_in_log(caplog, f"{some_body} sucessfully processed!", message_count_to_wait=1)
+    message_consumer.close()
+
+    received_heartbeat_frame_regex = re.compile(
+        f"Received frame.+heart-beat.+{settings.STOMP_OUTGOING_HEARTBIT},{settings.STOMP_INCOMING_HEARTBIT}"
+    )
+    sending_heartbeat_message_regex = re.compile("Sending a heartbeat message at [0-9.]+")
+    sending_ack_frame_regex = re.compile(f"Sending frame: .+ACK.+subscription:{message_consumer._subscription_id}.+")
+
+    assert any(received_heartbeat_frame_regex.match(m) for m in caplog.messages)
+    assert any(sending_heartbeat_message_regex.match(m) for m in caplog.messages)
+    assert any(sending_ack_frame_regex.match(m) for m in caplog.messages)
```

Comment:
How about this?
```suggestion
    assert any(received_heartbeat_frame_regex.match(m) for m in caplog.messages)
    assert any(sending_heartbeat_message_regex.match(m) for m in caplog.messages)
    assert any(sending_ack_frame_regex.match(m) for m in caplog.messages)
    assert not any(re.compile("Heartbeat loop ended").match(m) for m in caplog.messages)
```
Maybe the implementation is wrong considering this method. I copied it from [here](https://github.com/juntossomosmais/django-stomp/blob/121da40e6ffaca41e9d3a3e9041ccdc3914b0596/tests/integration/test_execution.py#L402).

Reply:
A better assertion, and definitely what we want, would be:

```suggestion
    assert any(received_heartbeat_frame_regex.match(m) for m in caplog.messages)
    assert any(sending_heartbeat_message_regex.match(m) for m in caplog.messages)
    assert any(sending_ack_frame_regex.match(m) for m in caplog.messages)
    assert not any(re.compile("Heartbeat timeout").match(m) for m in caplog.messages)
```

Reply:
A `Heartbeat loop ended` is just to inform that the thread that is dealing with heartbeats is not running anymore